script.js

```diff
@@ -1 +1,380 @@
+  }, delayInMilliseconds);
+};
+
+const initGame = () => {
+  // create this special deck by getting the doubled cards and
+  // making a smaller array that is ( boardSize squared ) number of cards
+  let doubleDeck = makeDeck();
+  let deckSubset = doubleDeck.slice(0, boardSize * boardSize);
+  deck = shuffleCards(deckSubset);
+
+  // deal the cards out to the board data structure
+  for (let i = 0; i < boardSize; i += 1) {
+    board.push([]);
+    for (let j = 0; j < boardSize; j += 1) {
+      board[i].push(deck.pop());
+    }
+  }
+
+  boardEl = buildBoardElements(board);
+  document.body.appendChild(boardEl);
+};
+
+initGame();
+
+//create reset Button
+const resetButton = document.createElement("button");
+resetButton.innerText = "Reset Game";
+resetButton.addEventListener("click", resetGame);
+resetButton.class = "reset";
+
+//create element for reset button
+const elementReset = document.createElement("div");
+elementReset.appendChild(resetButton);
+document.body.appendChild(elementReset);
+
+//function to display stopwatch time in hour, mins, sec
+const displayHMS = (millSecs) => {
+  const timeSecs = millSecs / 1000;
+  let minutes = Math.floor(timeSecs / 60);
+  const seconds = timeSecs % 60;
+  const hours = Math.floor(minutes / 60);
+  minutes %= 60;
+  const padSeconds = String(seconds).padStart(2, 0);
+  const padMinutes = String(minutes).padStart(2, 0);
+  const padHours = String(hours).padStart(2, 0);
+  return `${padHours}:${padMinutes}:${padSeconds}`;
+};
+
+//funtion to control the elapsed time
+const runStopWatch = () => {
+  const stopwatchCount = setInterval(() => {
+    stopwatchTime.innerHTML = `${displayHMS(milliSecs)}`;
+    if (isStopPressed) {
+      clearInterval(stopwatchCount);
+      milliSecs -= 1000;
+    }
+    milliSecs += 1000;
+  }, 1000);
+};
+
+//create Stopwatch elements
+const stopwatchTime = document.createElement("div");
+stopwatchTime.innerHTML = "00:00:00";
+stopwatchTime.classList.add("elapsedTime");
+document.body.appendChild(stopwatchTime);
+
+const stopwatchButtons1 = document.createElement("div");
+const stopwatchButtons2 = document.createElement("div");
+
+const stopwatchStart = document.createElement("button");
+stopwatchStart.innerHTML = "START";
+stopwatchButtons1.appendChild(stopwatchStart);
+stopwatchStart.classList.add("stopwatchButton");
+stopwatchStart.addEventListener("click", runStopWatch);
+
+const stopwatchStop = document.createElement("button");
+stopwatchStop.innerHTML = "STOP";
+stopwatchButtons1.appendChild(stopwatchStop);
+stopwatchStop.classList.add("stopwatchButton");
+stopwatchStop.addEventListener("click", () => {
+  isStopPressed = true;
+  runStopWatch();
+});
+
+const stopwatchReset = document.createElement("button");
+stopwatchReset.innerHTML = "RESET";
+stopwatchButtons2.appendChild(stopwatchReset);
+stopwatchReset.classList.add("stopwatchButton");
+stopwatchReset.addEventListener("click", () => {
+  milliSecs = 0;
+  isStopPressed = true;
+  runStopWatch();
```

Comment:
consider not running runStopWatch() here, instead, reset the timer display like this :
`stopwatchTime.innerHTML = "00:00:00";`
also consider adding a canClick variable so that the reset button cannot be clicked unless the stop button has already been clicked:
`let canClick = false;`